lib/Doctrine/ORM/Tools/Setup.php

```diff
@@ -157,19 +157,18 @@ private static function createCacheConfiguration(bool $isDevMode, string $proxyD
 
     private static function createCacheInstance(bool $isDevMode, ?Cache $cache) : Cache
     {
-        if ($cache !== null) {
```

Comment:
If a cache is given, it should be used. This change is a major BC break for anyone relying on current `$isDevMode` semantics